clientcontroller/babylon.go

```diff
@@ -174,8 +178,32 @@ func (bc *BabylonController) SubmitCovenantSigs(covSigs []*types.CovenantSigs) (
 			SlashingUnbondingTxSigs: covSig.SlashingUnbondingSigs,
 		})
 	}
+
+	return bc.reliablySendMsgsResendingOnMsgErr(msgs)
+}
+
+// reliablySendMsgsResendingOnMsgErr sends the msgs to the chain, if some msg fails to execute
+// and contains 'message index: %d', it will remove that msg from the batch and send again
+// if there is no more message available, returns the last error.
+func (bc *BabylonController) reliablySendMsgsResendingOnMsgErr(msgs []sdk.Msg) (*types.TxResponse, error) {
 	res, err := bc.reliablySendMsgs(msgs)
 	if err != nil {
+		// something failed, check if it is the message index failure
+		if res == nil || len(msgs) <= 1 {
+			return nil, err
+		}
+
+		if strings.Contains(err.Error(), "message index: ") {
+			// remove the failed msg from the batch and send again
+			failedIndex, found := FailedMessageIndex(err)
+			if !found {
+				return nil, err
+			}
+
+			newMsgs := RemoveMsgAtIndex(msgs, failedIndex)
+			return bc.reliablySendMsgsResendingOnMsgErr(newMsgs)
```

Comment:
I would avoid recursion here and do it in the loop. Recursive calls which communicate over network seems like asking ourself for stack overflow errors.

Reply:
addressed the comment, will wait for another approval to merge